src/libpriv/rpmostree-core.c

```diff
@@ -2926,6 +2961,9 @@ checkout_package_into_root (RpmOstreeContext *self,
                             GCancellable *cancellable,
                             GError      **error)
 {
+  g_auto(GStrv) files_remove_regex = 
+    ror_treefile_get_files_remove_regex (self->treefile_rs, dnf_package_get_name (pkg));
```

Comment:
`treefile_rs` is only used on the compose side, so we need to make this conditional on `self->treefile_rs`, e.g. something like:

```c
g_auto(GStrv) ... = NULL;
if (self->treefile_rs)
  ... = ror_treefile_get...
```

Reply:
Ah, that would explain why all the non-compose tests are failing. So basically `checkout_package_into_root()` is also called on the client side, but when it's called from the client side, `treefile_rs` is not available in the `RpmOstreeContext`.